src/commands/mod.rs

```diff
@@ -858,29 +868,32 @@ impl DaemonControl {
         is_cancel: bool,
         feerate_vb: Option<u64>,
     ) -> Result<CreateSpendResult, CommandError> {
-        // The transaction to be replaced must be one of ours and so we can assume it signals replaceability.
         let mut db_conn = self.db.connection();
 
+        if is_cancel && feerate_vb.is_some() {
+            return Err(CommandError::RbfError(RbfErrorInfo::SuperfluousFeerate));
+        }
+
         let prev_psbt = db_conn
             .spend_tx(txid)
             .ok_or(CommandError::UnknownSpend(*txid))?;
-        let prev_coins = {
-            let prev_outpoints: Vec<bitcoin::OutPoint> = prev_psbt
-                .unsigned_tx
-                .input
-                .iter()
-                .map(|txin| txin.previous_output)
-                .collect();
-            let prev_coins = db_conn.coins_by_outpoints(&prev_outpoints);
-            // Make sure all prev outpoints are coins in our DB.
-            if let Some(op) = prev_outpoints
-                .iter()
-                .find(|op| !prev_coins.contains_key(op))
-            {
-                return Err(CommandError::UnknownOutpoint(*op));
-            }
-            prev_coins
-        };
+        if !prev_psbt.unsigned_tx.is_explicitly_rbf() {
+            return Err(CommandError::RbfError(RbfErrorInfo::NotSignaling));
```

Comment:
Should we rather check all the spending transactions in the mempool? Maybe using the `bip125-replaceable` field in `getmempoolentry`?

Reply:
I think the directly conflicting transactions should correspond to the `mempool_spenders` for the previous outpoints, but happy to come back to that in a follow-up.